src/utils/getNolUrls.js

```diff
@@ -6,14 +6,16 @@ const getNolUrl = (course) => {
   const lang = "CH";
   const base_url = "https://nol.ntu.edu.tw/nol/coursesearch/print_table.php?";
   const course_id = course.identifier.replace("E", "");
+  const dept_id = course.departments.length > 0 ? course.departments[0].id : "";
+  const class_id = course.class ? course.class : "";
```

Comment:
Try to use optional chaining

```suggestion
  const dept_id =  course.departments?.[0]?.id ?? "";
  const class_id = course?.class ?? "";
```

Reply:
Resolved, thanks!